### PR TITLE
feat(replay): Update privacy docs for clarity

### PR DESCRIPTION
**Pull request by mydea**
* Clarifies that masking happens in the SDK
* Updates the types for the replay options to be code
* Adds config to the mask/block/ignore sections directly

Closes https://github.com/getsentry/sentry-docs/issues/8517

### CODE REVIEW

File: src/platform-includes/replay/privacy-configuration/javascript.mdx
Comment (lforst) [inline src/platform-includes/replay/privacy-configuration/javascript.mdx:5]: Can we make this also valid js (i.e. strings)? Same for the `block`, `unblock`, `ignore` and `unmask` variables below.

File: src/platform-includes/replay/privacy-configuration/javascript.mdx
Comment (lforst) [inline src/platform-includes/replay/privacy-configuration/javascript.mdx:11]: Why remove the `: string` here?

**Review by lforst — APPROVED** [review]
❤️

**Review by billyvg — APPROVED** [review]
lgtm aside from luca's comments

**Review by shanamatthews — APPROVED** [review]
Fine from my perspective!

File: src/platform-includes/replay/privacy-configuration/javascript.mdx
Comment (lforst) [reply src/platform-includes/replay/privacy-configuration/javascript.mdx:11]: If you ask me we should prioritize correctness over looks here 😄 But I'll leave it up to you.

**Pull request merged**
